ext/libxml/libxml.c

```diff
@@ -1071,29 +1063,44 @@ PHP_FUNCTION(libxml_disable_entity_loader)
 /* {{{ Changes the default external entity loader */
 PHP_FUNCTION(libxml_set_external_entity_loader)
 {
+	zval					*callback;
 	zend_fcall_info			fci;
 	zend_fcall_info_cache	fcc;
+	char					*error = NULL;
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_FUNC_OR_NULL(fci, fcc)
+		Z_PARAM_ZVAL(callback)
 	ZEND_PARSE_PARAMETERS_END();
 
-	_php_libxml_destroy_fci(&LIBXML(entity_loader).fci, &LIBXML(entity_loader).object);
-
-	if (ZEND_FCI_INITIALIZED(fci)) { /* argument not null */
-		LIBXML(entity_loader).fci = fci;
-		Z_ADDREF(fci.function_name);
-		if (fci.object != NULL) {
-			ZVAL_OBJ(&LIBXML(entity_loader).object, fci.object);
-			Z_ADDREF(LIBXML(entity_loader).object);
+	if (Z_ISNULL_P(callback)) {
+		LIBXML(entity_loader).fci.size = 0;
+	} else {
+		fci.size = sizeof(fci);
+		fci.object = NULL;
+		fci.named_params = NULL;
+		if (zend_fcall_info_init(callback, 0, &fci, &fcc, NULL, &error) != SUCCESS) {
+			zend_argument_type_error(1, "must be a valid callback or NULL, %s", error);
```

Comment:
Null should be written with lower case letters:
```suggestion
			zend_argument_type_error(1, "must be a valid callback or null, %s", error);
```

Reply:
Done